File: RockPaperScissors.kt
import java.util.*

fun main() {
    // Welcome the player to the game
    println("Welcome to Rock, Paper, Scissors!")

    // Define the choices
    val choices = arrayOf("rock", "paper", "scissors")

    // Prompt the player to make a choice
    print("Please choose one: rock, paper, or scissors ")
    var playerChoice = readLine()!!.toLowerCase()

    // Validate the player's choice
    while (playerChoice != "rock" && playerChoice != "paper" && playerChoice != "scissors") {
        println("Sorry, that's not a valid choice. Please choose one: rock, paper, or scissors")
        print("Please choose one: rock, paper, or scissors ")
        playerChoice = readLine()!!.toLowerCase()
    }

    // Generate the computer's choice
    val rand = Random()
    val computerChoice = choices[rand.nextInt(choices.size)]

    // Determine the winner
    if (playerChoice == computerChoice) {
        println("It's a tie!")
    } else if (playerChoice == "rock" && computerChoice == "scissors") {
        println("You win! Rock beats scissors.")
    } else if (playerChoice == "paper" && computerChoice == "rock") {
        println("You win! Paper beats rock.")
    } else if (playerChoice == "scissors" && computerChoice == "paper") {
        println("You win! Scissors beats paper.")
    } else {
        println("You lose. " + computerChoice.capitalize() + " beats " + playerChoice + ".")
    }
}

fun String.capitalize() = this[0].toUpperCase() + this.substring(1)

// To play the game, save the code to a file (e.g. RockPaperScissors.kt) and run it with the kotlin command in your terminal
// kotlin RockPaperScissors.kt 

File: rock_paper_scissors.py
# Welcome the player to the game
print("Welcome to Rock, Paper, Scissors!")

# Define the choices
choices = ["rock", "paper", "scissors"]

# Prompt the player to make a choice
player_choice = input("Please choose one: rock, paper, or scissors ").lower()

# Validate the player's choice
while player_choice not in choices:
  print("Sorry, that's not a valid choice. Please choose one: rock, paper, or scissors")
  player_choice = input("Please choose one: rock, paper, or scissors ").lower()

# Generate the computer's choice
import random
computer_choice = random.choice(choices)

# Determine the winner
if player_choice == computer_choice:
  print("It's a tie!")
elif player_choice == "rock" and computer_choice == "scissors":
  print("You win! Rock beats scissors.")
elif player_choice == "paper" and computer_choice == "rock":
  print("You win! Paper beats rock.")
elif player_choice == "scissors" and computer_choice == "paper":
  print("You win! Scissors beats paper.")
else:
  print(f"You lose. {computer_choice.capitalize()} beats {player_choice}.")

# To play the game, save the code to a file (e.g. rock_paper_scissors.py) and run it with the python command in your terminal
# python rock_paper_scissors.py
